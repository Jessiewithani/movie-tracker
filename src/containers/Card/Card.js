import React from 'react';
import '../Card/Card.scss'
import images from '../../assets/images';

const Card = ({ movie, poster, title, favorited, toggleFavorites }) => {

  return (
    <div className="card">
      <img className="favorite-icon" src={images.favIcon} alt="" onClick={(e) => toggleFavorites(e, movie)} />
      <label className="favorite-label">FAVORITE</label>
      <img className="card-img" src={poster} alt="movie poster" />
      <h3 className ="movie-title">{title}</h3>
    </div>
  )
}

export default Card;