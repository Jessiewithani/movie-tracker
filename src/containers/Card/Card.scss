.card {
    margin: 10vh 0 5vh 0;
    margin-left: 20px;
    background-color: rgba(15, 30, 45, 0.8);
    ;
    color: white;
    border-radius: 5px;
}

.movie-title {
    height: 50px;
    letter-spacing: 1.5px;
}

.card-img {
    height: 500px;
    width: auto;
    margin-bottom: 25px;
}

h5 {
    height: 30px;
}

.favorite-icon {
    height: 50px;
    margin: 18px 8px;
    position: absolute;
    border-radius: 50%;
}

.favorite-icon:hover {
    box-shadow: 0px 0px 9px 4px rgb(46, 197, 147);
    animation: glow 1.5s linear infinite alternate;
}

.favorite-icon:active {
    box-shadow: 0px 0px 8px 4px rgb(224, 232, 116);

}

@keyframes glow {
    to {
        box-shadow: 0px 0px 20px 14px rgb(46, 197, 147);
    }
}

.favorite-label {
    font-size: 14px;
    font-weight: 500;
    position: absolute;
    border-radius: 6px;
    background: rgba(204, 69, 69, 0.479);
}