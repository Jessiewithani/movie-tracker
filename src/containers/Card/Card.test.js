import React from 'react';
import Card from './Card';
import { shallow } from 'enzyme';

describe('Card', () => {
  let wrapper;
  beforeEach(() => {
     wrapper = shallow(
      <Card 
      key={4}
      poster='https://image.tmdb.org/t/p/w342$'
      title='Batman'
      favorited='false'
      />
    )
  })
  it('should match the snapshot with with all data passed in', () => {
    expect(wrapper).toMatchSnapshot()
  })
})