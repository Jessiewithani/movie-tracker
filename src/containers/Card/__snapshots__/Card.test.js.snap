// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot with with all data passed in 1`] = `
<div
  className="card"
>
  <img
    alt="movie poster"
    className="card-img"
    src="https://image.tmdb.org/t/p/w342$"
  />
  <h3
    class="movie-title"
  >
    Batman
  </h3>
</div>
`;
